packages/scalar.aspnetcore/ScalarOptions.cs

```diff
@@ -28,6 +28,12 @@ public class ScalarOptions
     public Dictionary<string, string>? Metadata { get; set; }
 
     public ScalarAuthenticationOptions? Authentication { get; set; }
+
+    /// <summary>
+    /// Provides a function for setting the data URL for the API reference
+    /// given the document name.
+    /// </summary>
+    public Func<string, string>? DataUrlFunc { get; set; }
```

Comment:
What about another name like `DataUrlProvider`/`DateUrlFactory`?

Reply:
`Provider` and `Factory` suffixes, IMO, indicate that the type you might expect to pass is more complex (e.g. some class implementation). Something like `GenerateDataUrl` might work as well.